rholang/src/test/scala/coop/rchain/rholang/StackSafetySpec.scala

```diff
@@ -71,11 +90,17 @@ class StackSafetySpec extends FlatSpec with Matchers {
   }
 
   it should "handle a huge nested list" in {
-    checkNormalize(hugeNested("[", "", "]"))
+    checkAll(hugeNested("[", "", "]"))
   }
 
+  //FIXME java.lang.OutOfMemoryError: GC overhead limit exceeded
```

Comment:
Do you think this could be related to the way you calculate `depth`?

Reply:
It's been there way before I had depth calculation in place. I suppose this is due to each level of the set normalizing everything that's being put inside, resulting in quadratic waste.